go/vt/vtgate/planbuilder/operators/query_planning.go

```diff
@@ -107,6 +107,13 @@ func runRewriters(ctx *plancontext.PlanningContext, root Operator) Operator {
 		}
 	}
 
+	if pbm, ok := root.(*PercentBasedMirror); ok {
```

Comment:
if i don't do this, then the source and target of the mirror will be seed the operator list without taking mirroring into account. not sure if there's a better way to get `ctx.UseMirror()` into the `seedOperatorList` of the target-side mirror.

Reply:
an alternative to this would be to place the mirror tables into the `QueryGraph` when it is constructed here: https://github.com/vitessio/vitess/blob/f73f62468e9852be321139110b6758c98fdb60c6/go/vt/vtgate/planbuilder/operators/ast_to_op.go#L240-L262

if we did that, then we might be able to delete `plancontext/mirror_vschema.go`.

here's what it might look like:

```diff
diff --git a/go/vt/vtgate/planbuilder/operators/ast_to_op.go b/go/vt/vtgate/planbuilder/operators/ast_to_op.go
index 5cf76ee498..9fe3a3e043 100644
--- a/go/vt/vtgate/planbuilder/operators/ast_to_op.go
+++ b/go/vt/vtgate/planbuilder/operators/ast_to_op.go
@@ -276,6 +276,16 @@ func getOperatorFromAliasedTableExpr(ctx *plancontext.PlanningContext, tableExpr
 		}
 		qg := newQueryGraph()
 		isInfSchema := tableInfo.IsInfSchema()
+		if mr := tableInfo.GetMirrorRule(); mr != nil {
+			newTbl := sqlparser.Clone(tbl)
+			newTbl.Qualifier = sqlparser.NewIdentifierCS(mr.Table.Keyspace.Name)
+			newTbl.Name = mr.Table.Name
+			if newTbl.Name.String() != tbl.Name.String() {
+				tableExpr = sqlparser.Clone(tableExpr)
+				tableExpr.As = tbl.Name
+			}
+			tbl = newTbl
+		}
 		qt := &QueryTable{Alias: tableExpr, Table: tbl, ID: tableID, IsInfSchema: isInfSchema}
 		qg.Tables = append(qg.Tables, qt)
 		return qg
```

not sure which is better.

on the one hand i don't like this because it duplicates logic already over here: https://github.com/vitessio/vitess/blob/f73f62468e9852be321139110b6758c98fdb60c6/go/vt/vtgate/planbuilder/operators/route.go#L445

but on the other hand it would be nice delete `plancontext/mirror_vschema.go`

Reply:
either way is OK by me. you are right - neither alternative is perfect

Reply:
I did not like this `plancontext/mirror_vschema.go` file, the proposed change is not that bad.

Reply:
i will look into removing `plancontext/mirror_vschema.go` and try the proposed change instead.

Reply:
was able to implement that proposal and remove `mirror_vschema.go`. i like it better this way.